### PR TITLE
Regular deploys of Balrog & Ship It

### PR DESCRIPTION
**Pull request by bhearsum**
[rendered](https://github.com/mozilla-releng/releng-rfcs/blob/5ddb8281d1507b3b66ef540c49473fd17a422a7c/rfcs/0043-regular-deploys.md)

### CODE REVIEW

**Comment by gabrielBusta** [comment]
Thought - should we also do s-s? We could batch that with shipit.

RE the front-end dependencies on shipit and balrog I could take a stab at it.

**Comment by bhearsum** [comment]
> Thought - should we also do s-s? We could batch that with shipit.

I can definitely see an argument for doing scriptworker-scripts too, yeah! I totally forgot about those when writing this, but they are obviously the other major piece of infra.

> RE the front-end dependencies on shipit and balrog I could take a stab at it.

Can you clarify what you mean here? Even if we were to do a one time update of the dependencies (which is a good idea IMO), doing regular updates after that still carries higher bustage risk due to lack of tests.

**Comment by gabrielBusta** [comment]
> Can you clarify what you mean here? Even if we were to do a one time update of the dependencies (which is a good idea IMO), doing regular updates after that still carries higher bustage risk due to lack of tests.

Oh I see. I was thinking on a tangent about updating React/MUI to the next major version (I see they are still getting security patches so it's not to pressing.) Tests would come in handy :/

**Comment by bhearsum** [comment]
> > Can you clarify what you mean here? Even if we were to do a one time update of the dependencies (which is a good idea IMO), doing regular updates after that still carries higher bustage risk due to lack of tests.
> 
> Oh I see. I was thinking on a tangent about updating React/MUI to the next major version (I see they are still getting security patches so it's not to pressing.) Tests would come in handy :/

I think that's a good thing to do regardless! I'm just not sure doing so would affect anything here (but I could be convinced otherwise).

File: rfcs/0043-regular-deploys.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+Few changes will be needed to implement this proposal. We already have tools and processes for updating dependencies and deploying new versions - the main thing needed will be reminders for those on releaseduty. To this end, we will add two new Slack notifications targeting @releng:
+
+1) A reminder 2 days before the expected deploy date to update the dependencies and deploy to stage. This will give us time to sanity check and ideally fix any issues.
+2) A reminder on deploy day to perform the production deployment.
+
+The other change we will make as part of this is to ensure that RelEng can complete Balrog deployments without the involvement of the SRE team. (This is not strictly necessary to get where we want, but it avoids extra busy work for SREs, and moves in the direction that they would like to go anyways.)
+
+## Schedule
+
+We will do these deploys during the same weeks as Firefox CI Taskcluster deploys, but on different days. Taskcluster deploys happen on Wednesdays, so we should stage these deployments on Tuesdays, and push to production on Thursdays.
```
Comment (ahal) [inline rfcs/0043-regular-deploys.md:28]: Firefox-CI Taskcluster deploys are going to be moving to Thursdays due to the new beta schedule. Maybe we should just do these the opposite week as fxci deploys?

File: rfcs/0043-regular-deploys.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+# Open Questions
+
+* Are we happy with the proposed schedule above?
+* Will we grant RelEng the necessary access in Jenkins to click the deploy button, or remove the need for SREs in some other way?
+* Should we update backend and frontend dependencies? Or just backend? (Frontend upgrades carry more risk because we have no tests for this code.)
```
Comment (ahal) [inline rfcs/0043-regular-deploys.md:34]: Maybe just backend for now until we are more confident with front-end?

**Review by ahal — APPROVED** [review]
I agree with this proposal. Do we want to include scriptworker-scripts? I think more deploys in general isn't necessarily a bad thing.

**Comment by bhearsum** [comment]
> Do we want to include scriptworker-scripts? I think more deploys in general isn't necessarily a bad thing.

Yeah, I think we should. I'll update the proposal to reflect this.

File: rfcs/0043-regular-deploys.md (hunk 1 above)
Comment (bhearsum) [reply rfcs/0043-regular-deploys.md:28]: Yes - that's a great idea!

**Pull request merged**